### PR TITLE
Add a first pass at a modal refill

### PR DESCRIPTION
**Pull request by creuter**
modal

* Closes when pressing escape, clicking the close button, or clicking the overlay
* focuses into the modal when opening
* focuses back to the open button when closing

### CODE REVIEW

**Comment by creuter** [comment]
I'd like to focus into the dialog when opened, but I'm not sure where to put focus

**Comment by drapergeek** [comment]
Code LGTM

File: source/stylesheets/refills/_modal.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  background-color: rgba($dark-gray, 0.75);
+}
+
+.modal {
+  @include position(fixed, 50% null null 50%);
+  background-color: $base-background-color;
+  max-width: calc(100% - #{$base-spacing} * 2);
+  overflow: auto;
+  padding: $base-spacing;
+  transform: translate(-50%, -50%);
```
Comment (whmii) [inline source/stylesheets/refills/_modal.scss]: I wonder if we should just use flexbox here

Comment (creuter) [inline source/stylesheets/refills/_modal.scss]: Any thoughts on what advantages that would give us?

**Comment by creuter** [comment]
@tysongach Yeah, I'm not sure either. Even when explicitly focusing onto the title, it's not reading it, and I'm not sure what I'm missing.

modal-title-focus

Mindpatterns follows the same pattern I did here: http://ianmcburnie.github.io/mindpatterns/disclosure/dialog/index.html

**Pull request merged**